ogc-common/src/main/java/io/arlas/server/ogc/common/utils/XmlUtils.java

```diff
@@ -68,47 +71,38 @@ public static void parsePropertiesXsd(Map<String, CollectionReferenceDescription
-                        }else{
-                            DateTimeFormatter dtf = null;
-                            TimestampType type = TimestampType.getElasticsearchPatternName(property.format);
-                            if (type.name().equals("UNKNOWN")) {
-                                dtf = DateTimeFormat.forPattern(property.format);
-                            } else {
-                                dtf = type.dateTimeFormatter;
-                            }
-                            if (dtf != null) {
-                                DateTime jodatime = dtf.parseDateTime((String)valueObject);
-                                Long value = jodatime.getMillis();
-                                String timestamp = "" + value;
-                                writeElement(writer, String.join(".", new ArrayList<>(namespace)), f.format(new Date(Long.parseLong(timestamp))), uri, prefix);
-                            }
-                        }
-                    }
-                }
-            }
-            namespace.pop();
-        }
-    }
+           if (properties != null) {
+               for (String key : properties.keySet()) {
+                   CollectionReferenceDescriptionProperty property = properties.get(key);
+                   namespace.push(key);
+                   String path = String.join(".", new ArrayList<>(namespace));
+                   boolean excludePath = excludeFields.stream().anyMatch(pattern -> pattern.matcher(path).matches());
+                   if (!excludePath) {
+                       if (property.type == ElasticType.OBJECT) {
+                           parsePropertiesXml(property.properties, writer, namespace, uri, source, prefix, excludeFields);
+                       } else {
+                           Object valueObject = MapExplorer.getObjectFromPath(String.join(".", new ArrayList<>(namespace)), source);
+                           if (valueObject != null && property.type != ElasticType.DATE && property.type != ElasticType.GEO_POINT && property.type != ElasticType.GEO_SHAPE) {
+                               String value = valueObject.toString();
+                               writeElement(writer, String.join(".", new ArrayList<>(namespace)), value, uri, prefix);
+                           }else if(valueObject != null && property.type == ElasticType.DATE ){
+                               SimpleDateFormat f = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSXX");
+                               f.setTimeZone(TimeZone.getTimeZone("UTC"));
+                               if (property.format == null) {
+                                   property.format = CollectionReference.DEFAULT_TIMESTAMP_FORMAT;
+                               }
+                               Long timestamp = TimestampTypeMapper.getTimestamp(valueObject,property.format);
```

Comment:
Not sure the timestamp variable is used

Reply:
my bad, I didn't put the right variable. I fixed it